core/src/main/scala/pureconfig/DerivedWriters.scala

```diff
@@ -108,8 +119,27 @@ trait DerivedWriters1 {
     }
   }
 
+  // Auxiliary type to make sure we always return `ConfigList` when writing HLists.
+  private[pureconfig] trait HListConfigWriter[T <: HList] {
```

Comment:
When I originally wrote this we had `WrappedConfigWriter` extending `ConfigWriter` which provided a conflicting way of writing `HLists` as empty `ConfigObject`s. I ended up working around it by introducing this `HListConfigWriter`. 

In the end I also had to make `WrappedConfigWriter` not extending `ConfigWriter` as otherwise we couldn't derive a proper `ConfigWriter` for `HNil`. In that case, I can make this extend `ConfigWriter` and make those methods public. I've done that in 3a758ff826f72c37c2cf2a10eba18027cdfea51c.